MuchToDoWithoutAdo/app/src/main/java/com/scottlindley/muchtodowithoutado/RecyclerViewAdapters/ListScreenRecyclerViewAdapter.java

```diff
@@ -0,0 +1,221 @@
+        holder.mItemDescription.setText(mItems.get(holder.getAdapterPosition()).mItemDescription);
+        holder.mCheckBox.setChecked(mItems.get(holder.getAdapterPosition()).mIsChecked);
+        holder.mCheckBox.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (!mItems.get(holder.getAdapterPosition()).mIsChecked) {
+                    mItems.get(holder.getAdapterPosition()).mIsChecked = true;
+                    notifyItemChanged(holder.getAdapterPosition());
+                } else {
+                    mItems.get(holder.getAdapterPosition()).mIsChecked = false;
+                    notifyItemChanged(holder.getAdapterPosition());
+                }
+            }
+        });
+
+        holder.mRelativeLayout.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                holder.mInfoButton.setVisibility(View.VISIBLE);
+                holder.mInfoButton.setClickable(true);
+                holder.mInfoButton.postDelayed(new Runnable() {
+                    @Override
+                    public void run() {
+                        holder.mInfoButton.setVisibility(View.INVISIBLE);
+                        holder.mInfoButton.setClickable(false);
+                    }
+                }, 2000);
+            }
+        });
+
+        holder.mRelativeLayout.setOnLongClickListener(new View.OnLongClickListener() {
+            @Override
+            public boolean onLongClick(View view) {
+                mItems.remove(holder.getAdapterPosition());
+                notifyItemRemoved(holder.getAdapterPosition());
+                return true;
+            }
+        });
+
+        holder.mInfoButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                final AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(view.getContext());
+
+                LayoutInflater inflater = LayoutInflater.from(view.getContext());
+                final View dialogView = (inflater.inflate(R.layout.edit_list_item_dialog, null));
+                dialogBuilder.setView(dialogView);
+                final EditText itemName = (EditText) dialogView.findViewById(R.id.new_todo);
+                final EditText itemDescription = (EditText) dialogView.findViewById(R.id.details);
+                final RadioButton lowPriority = (RadioButton) dialogView.findViewById(R.id.low_priority);
+                final RadioButton mediumPriority = (RadioButton) dialogView.findViewById(R.id.medium_priority);
+                final RadioButton highPriority = (RadioButton) dialogView.findViewById(R.id.high_priority);
+
+                itemName.setHint(mItems.get(holder.getAdapterPosition()).mItemName);
+                itemDescription.setHint(mItems.get(holder.getAdapterPosition()).mItemDescription);
```

Comment:
If the user already supplied a value for the name and/or description, then call `setText()` to put that value in the EditText, rather than `setHint()`. If you make it a hint, then the text is still technically blank and you can end up getting an error saying name can't be empty even though it looks like it's not empty.